Import-time NameError in azuredevops_provider when Azure deps missing (breaks non-Azure providers too)

### Git provider

Github Enterprise

### System Info

PR-Agent version: v0.31 (release tag)
Python: 3.12.x
Deployment: Dockerized AWS service (EC2)
Git provider configured: GitHub (GitHub App)
Azure DevOps dependencies: NOT installed intentionally (we only use GitHub)

### Bug details

We run PR-Agent as a dockerized AWS service and use GitHub as the git provider. We do not install Azure DevOps dependencies in the image (to keep the runtime minimal), since Azure is not used.

However, PR-Agent fails at import-time with a NameError coming from `pr_agent/git_providers/azuredevops_provider.py` when the optional Azure DevOps dependencies are missing (or fail to import). This prevents PR-Agent from starting even though the configured git provider is GitHub.

Expected behavior
- GitHub deployments should start and run without requiring Azure DevOps dependencies.
- Azure provider should only error when the Azure provider is selected/instantiated.

Actual behavior
- Importing PR-Agent providers crashes due to type annotations referencing Azure classes that are not defined when Azure imports fail.

Why this happens
- `pr_agent/git_providers/__init__.py` imports `AzureDevopsProvider` unconditionally.
- In `pr_agent/git_providers/azuredevops_provider.py` Azure imports are wrapped in `try/except ImportError`, setting `AZURE_DEVOPS_AVAILABLE = False`.
- But method annotations like `-> Comment` and parameter annotations like `comment: Comment` still reference Azure types; when Azure imports fail, those names are undefined and cause `NameError` at module import time.

Minimal reproduction (docker)
1) Build/run a PR-Agent image WITHOUT Azure packages installed (azure-devops / azure-identity / msrest)
2) Start PR-Agent (or simply import providers)

Example:
- `python -c "import pr_agent.git_providers"`

Observed traceback:
 ```
 NameError: name 'Comment' is not defined
 def reply_to_comment_from_comment_id(...) -> Comment:
 File "/app/pr_agent/git_providers/azuredevops_provider.py", line 95, in AzureDevopsProvider
 File "/app/pr_agent/git_providers/__init__.py", line 4, in <module>
 from pr_agent.git_providers.azuredevops_provider import AzureDevopsProvider
 File "/app/pr_agent/servers/github_app.py", line 15, in <module>
 [ERROR] Exception in worker process
 [ERROR] Reason: Worker failed to boot.
```

Proposed fix (low-risk)
- Add `from __future__ import annotations` at the top of `pr_agent/git_providers/azuredevops_provider.py` to avoid evaluating annotations at import-time.
  Alternatives: quote Azure-only annotations (e.g. `-> "Comment"`) or guard imports via `TYPE_CHECKING`.

Impact
- This blocks PR-Agent startup in non-Azure environments where Azure dependencies are intentionally omitted, even when using GitHub.

--- a/pr_agent/git_providers/azuredevops_provider.py
+++ b/pr_agent/git_providers/azuredevops_provider.py
@@ -1,3 +1,5 @@
+from __future__ import annotations
+
 import os
 from typing import Optional, Tuple
 from urllib.parse import urlparse

